plugins/vsphere/pkg/command/source/create.go

```diff
@@ -0,0 +1,141 @@
+
+# Create the source in the specified namespace, sending events to the specified service with custom checkpoint behavior
+kn vsphere source create --namespace ns --name vc-01-source --address https://my-vsphere-endpoint.local --skip-tls-verify --secret-ref vsphere-credentials --sink-api-version v1 --sink-kind Service --sink-name the-service-name --checkpoint-age 1h --checkpoint-period 30s
+`,
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			if opts.Name == "" {
+				return fmt.Errorf("'name' requires a nonempty name provided with the --name option")
+			}
+			if opts.Address == "" {
+				return fmt.Errorf("'address' requires a nonempty address provided with the --address option")
+			}
+			if opts.SecretRef == "" {
+				return fmt.Errorf("'secret-ref' requires a nonempty secret reference provided with the --secret-ref option")
+			}
+			sinkCoordinatesAllEmpty := opts.SinkAPIVersion == "" && opts.SinkKind == "" && opts.SinkName == ""
+			sinkCoordinatesAllSet := opts.SinkAPIVersion != "" && opts.SinkKind != "" && opts.SinkName != ""
+			if opts.SinkURI == "" && sinkCoordinatesAllEmpty ||
+				(!sinkCoordinatesAllEmpty && !sinkCoordinatesAllSet) {
+				return fmt.Errorf("sink requires an URI" +
+					"\nand/or a nonempty API version --sink-api-version option," +
+					"\nwith a nonempty kind --sink-kind option," +
+					"\nand with a nonempty name with the --sink-name")
+			}
+
+			// verify supported datacontentencoding schemes
+			validEncodings := map[string]struct{}{
+				"xml":  {},
+				"json": {},
+			}
+			if _, ok := validEncodings[strings.ToLower(opts.PayloadEncoding)]; !ok {
+				return fmt.Errorf("invalid encoding scheme %q", opts.PayloadEncoding)
+			}
+
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			namespace, err := clients.GetExplicitOrDefaultNamespace(opts.Namespace)
+			if err != nil {
+				return fmt.Errorf("failed to get namespace: %v", err)
+			}
+			address, err := url.Parse(opts.Address)
+			if err != nil {
+				return fmt.Errorf("failed to parse source address: %v", err)
+			}
+			sinkDestination, err := opts.AsSinkDestination(namespace)
+			if err != nil {
+				return fmt.Errorf("failed to parse sink address: %v", err)
+			}
+			if _, err = clients.VSphereClientSet.
+				SourcesV1alpha1().
+				VSphereSources(namespace).
+				Create(cmd.Context(), newSource(namespace, sinkDestination, address, *opts), metav1.CreateOptions{}); err != nil {
+				return fmt.Errorf("failed to create source: %v", err)
+			}
+
+			fmt.Fprintln(cmd.OutOrStdout(), "Created source")
+			return nil
+		},
+	}
+
+	flags := result.Flags()
+	flags.StringVar(&opts.Name, "name", "", "name of the source to create")
+	flags.StringVarP(&opts.Address, "address", "a", "", "URL of ESXi or vCenter instance to connect to (same as VC_URL)")
```

Comment:
Maybe call this `vcenter` or `vc-url`? I confused this with `sink-uri` when reading some other code.

Reply:
Agree, will change.